--- gdb-9.2-new/gdb/windows-nat.c	2020-07-11 20:13:30.000000000 +0800
+++ gdb-9.2/gdb/windows-nat.c	2020-09-29 07:13:24.027344100 +0800
@@ -2736,20 +2736,20 @@
 	si.hStdInput = (HANDLE) _get_osfhandle (fd_inp);
       else if (tty != INVALID_HANDLE_VALUE)
 	si.hStdInput = tty;
-      else
-	si.hStdInput = GetStdHandle (STD_INPUT_HANDLE);
+      else 
+	si.hStdInput = NULL;
       if (fd_out >= 0)
 	si.hStdOutput = (HANDLE) _get_osfhandle (fd_out);
       else if (tty != INVALID_HANDLE_VALUE)
 	si.hStdOutput = tty;
       else
-	si.hStdOutput = GetStdHandle (STD_OUTPUT_HANDLE);
+	si.hStdOutput = NULL;
       if (fd_err >= 0)
 	si.hStdError = (HANDLE) _get_osfhandle (fd_err);
       else if (tty != INVALID_HANDLE_VALUE)
 	si.hStdError = tty;
       else
-	si.hStdError = GetStdHandle (STD_ERROR_HANDLE);
+	si.hStdError = NULL;
       si.dwFlags |= STARTF_USESTDHANDLES;
     }
 
